pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -2025,7 +2025,7 @@ private boolean shouldTopicBeRetained() {
         long retentionTime = TimeUnit.MINUTES.toNanos(retentionPolicies.getRetentionTimeInMinutes());
         // Negative retention time means the topic should be retained indefinitely,
         // because its own data has to be retained
-        return retentionTime < 0 || (System.nanoTime() - lastActive) < retentionTime;
+        return retentionTime <= 0 || (System.nanoTime() - lastActive) < retentionTime;
```

Comment:
If the `retentionTime` is `0`, we should not retain the data for the topic

Reply:
OK, the status will like this:
null: not retained
= 0: not retained
<0: retained indefinitely
> 0: retained as required